include/xsimd/arch/xsimd_neon.hpp

```diff
@@ -834,6 +834,21 @@ namespace xsimd
 
         namespace detail
         {
+            #define BITWISE_OR_TYPE(DataType, ParamType)                         \
+                inline ParamType bitwise_or_##DataType(ParamType a, ParamType b) \
+                {                                                                \
+                    return vorrq_##DataType(a, b);                               \
+                }
+            BITWISE_OR_TYPE(u8, uint8x16_t)
+            BITWISE_OR_TYPE(s8, int8x16_t)
+            BITWISE_OR_TYPE(u16, uint16x8_t)
+            BITWISE_OR_TYPE(s16, int16x8_t)
+            BITWISE_OR_TYPE(u32, uint32x4_t)
+            BITWISE_OR_TYPE(s32, int32x4_t)
+            BITWISE_OR_TYPE(u64, uint64x2_t)
+            BITWISE_OR_TYPE(s64, int64x2_t)
+            #undef BITWISE_OR_TYPE
+
```

Comment:
Defines a bunch of thin wrappers around vorrq_xxx intrinsics.
@JohanMabille , do you think it's okay? or any better approach?

Reply:
@cyb70289 the pattern is used through this file, so it's probably a good idea to have a more generic approach. What do you think of the following snippet:

https://godbolt.org/z/MoGPqWoqz

Reply:
The problem is that we cannot pass intrinsics to any function. In gcc -O0 or O1, it will take the passed intrinsics function pointer as normal function pointer and try to find the symbol at link time.

Please see https://godbolt.org/z/PG79js33a

`make_wrapped_functions(_mm_or_si128)`  (line 25) generates below code.
```
_GLOBAL__sub_I_t:
        mov     QWORD PTR t[rip], OFFSET FLAT:_Z12_mm_or_si128Dv2_xS_
        ret
```
It returns address of a symbol `_mm_or_si128` that gcc thinks exists somewhere. It will fail at link time as there's no such function implementation.

If you uncomment line 26 (and comment line 25), which enables a direct wrapping function,  the code is okay. The intrinsic assembly `por xmm0, xmm1` is inside the generated code.
```
my_or128(long long __vector(2), long long __vector(2)):
        por     xmm0, xmm1
        ret
_GLOBAL__sub_I_t:
        mov     QWORD PTR t[rip], OFFSET FLAT:my_or128(long long __vector(2), long long __vector(2))
        ret
```